File: api/waitlist/sse.py
from flask import Blueprint, g, redirect
from . import auth
from .webutil import ViewReturn
from .data import sse

bp = Blueprint("sse", __name__)


@bp.route("/api/sse/stream")
@auth.login_required
def sse_stream() -> ViewReturn:
    buckets = []
    buckets.append("waitlist")
    buckets.append("account;%d" % g.account_id)
    if auth.has_access("waitlist-view"):
        buckets.append("xup")

    return redirect(sse.get_events_url(buckets))

File: backend/src/routes/sse.rs
use rocket::response::Redirect;

use crate::core::auth::AuthenticatedAccount;

#[get("/api/sse/stream")]
fn stream(app: &rocket::State<crate::app::Application>, account: AuthenticatedAccount) -> Redirect {
    let mut topics = vec![
        "announcments".to_string(),
        "waitlist".to_string(),
        format!("account;{}", account.id),
    ];

    if account.access.contains("fleet-view") {
        topics.push("fleet_comp".to_string());
    }

    Redirect::temporary(app.sse_client.events_url(&topics))
}

pub fn routes() -> Vec<rocket::Route> {
    routes![stream,]
}
